v3io/dataplane/request.py

```diff
@@ -117,13 +117,19 @@ def encode_get_object(container_name, access_key, kwargs):
 
 
 def encode_put_object(container_name, access_key, kwargs):
-    headers = None
+    headers = {
+        "Content-Type": "application-octet-stream",
+    }
 
     # if the append flag is passed, add a range header
     if kwargs["append"]:
         headers = {"Range": "-1"}
 
-    return _encode("PUT", container_name, access_key, kwargs["path"], None, headers, kwargs["body"])
+    body = kwargs["body"]
+    if isinstance(body, str):
+        body = body.encode("utf-8")
+
+    return _encode("PUT", container_name, access_key, kwargs["path"], None, headers, body)
```

Comment:
You change the type of 'body' here from str to bytes. Does the caller of 'get' always know to convert bytes to str back?

Reply:
This PR does not affect the data written (see the description).

Reply:
Do notice that the operation here is put object. Objects don't have a type. They're always binary. Instead of `http.client` encoding string bodies as `latin-1`, we encode them as `utf-8`.